enhancements/baremetal/debuggability-of-baremetal-ipi-deployment.md

```diff
@@ -0,0 +1,246 @@
+   functional cluster)?
+2. In order to bubble up information about worker failures to the
+   installer, the most likely solution to me seems to use operator
+   status as Degraded, with a relevant error message. Could we mark
+   machine-api-operator when workers fail to roll out?
+
+## Summary
+
+In OpenShift 4.5, we improved the existing installer validations for
+baremetal IPI to identify early problems. Those include identifying
+duplicate baremetal host records, insufficient hardware resources to
+deploy the requested cluster size, reachability of RHCOS images, and
+networking misconfiguration such as overlapping networks or DNS
+misconfiguration.
+
+However, a variety of situations exist where deployments fail for
+reasons that were not preventable during the pre-install validations.
+These failures in baremetal IPI are hard to diagnose. Errors from
+baremetal-operator and ironic are often not presented to the user, and
+even when they are the installer doesn't provide context about what
+action to take.
+
+This enhancement request is a broad attempt at categorizing the types of
+deployment failures, and what information we could present to the user
+to make identifying root causes easier.
+
+## Motivation
+
+The goal of this enhancement is to improve the day 1 install experience
+and reduce the perception of complexity in baremetal IPI deployments.
+
+### Goals
+
+- Any deployment that ends in an unsuccessful install must provide the
+  user clear and actionable information to diagnose the problem.
+
+### Non-Goals
+
+- Addressing the underlying causes of the failures are not the goal of
+  this enhancement.
+
+## Proposal
+
+Broadly, deployments fail due to problems encountered during these
+installation activities:
+
+  - Pre-bootstrap (image downloading, manifest creation, etc)
+  - Infrastructure automation (Terraform)
+  - Bootstrap
+  - Bare Metal Host Provisioning (Control Plane and Workers)
+  - Operator Deployment (i.e., those rolled out by CVO)
+
+We believe that since 4.5, pre-bootstrap errors are usually detected,
+and useful information is presented to the user about how to rectify the
+problem, so this enhancement request will focus on failures that occur
+from terraform onward.
+
+### Kinds of deployment failure
+
+#### Infrastructure Automation (Terraform)
+
+Baremetal IPI relies on terraform to provision a libvirt bootstrap
+virtual machine, and bare metal control plane hosts. We use
+terraform-provider-libvirt and terraform-provider-ironic to accomplish
+those goals.
+
+terraform-provider-ironic reports failures when it cannot reach the
+Ironic API, or a control plane host fails to provision. In both cases,
+we do not provide useful information to the user about what to do.
+
+#### Bootstrap Failures
+
+The bootstrap runs a couple of baremetal-specific services, including
+Ironic as well as a utility that populates introspection data for
+the control plane.
```

Comment:
```suggestion
Ironic and a utility that populates hardware information for
the control plane hosts.
```

Calling out "hardware details" is probably more clear for those not already familiar with Ironic terminology